lib/fmrest/spyke/relation.rb

```diff
@@ -268,6 +270,35 @@ def or(*params)
         params.empty? ? clone : clone.query(*params)
       end
 
+      def and(*params)
+        params = params.map { |p| normalize_query_params(p) }
+
+        if query_params.including(params).any? { |param| param.key?("omit") }
+          raise ArgumentError, "Cannot use `and' with `omit'"
+        end
+
+        clone = with_clone do |r|
+          if r.query_params.empty?
+            r.query_params = params
+          else
+            r.query_params =
+              r
+                .query_params
+                .product(params)
+                # A field cannot hold two values at the same time, so mark
+                # those cases for removal:
+                .map { |a, b| a.merge(b) { |_, v1, v2| v1 == v2 ? v1 : ImpossibleValue } }
```

Comment:
@marquete I changed `nil` to this custom `ImpossibleValue` here because even though `nil` doesn't have any special meaning in FmRest, I could see us using it for querying in some way down the line, so I prefer to use a special value to mark these so we don't accidentally remove valid query values from the request.